artiq/frontend/artiq_dashboard.py

```diff
@@ -113,10 +113,10 @@ def main():
         rpc_clients[target] = client
 
     config = Client(args.server, args.port_control, "master_config")
-    atexit.register(config.close_rpc)
-    server_name = config.get_name()
-    if server_name is None:
-        server_name = args.server
+    try:
+        server_name = config.get_name()
+    finally:
+        config.close_rpc()
 
```

Comment:
I assume this is what you were looking for? Client() tidies up after itself if it has an exception, so the only logic that needs guarding seems to be the RPC.

Reply:
Yes.